From 85f1cf2f74e2f5dc18a57eded07faed148e4fec2 Mon Sep 17 00:00:00 2001
From: Christophe Fergeau <cfergeau@redhat.com>
Date: Wed, 16 Apr 2014 13:56:48 +0200
Subject: [PATCH] ovirt: Remove extra '/' from oVirt URI

The 'path' part of the URI will always start with a '/' when present as
this is what separates it from the hostname. When rebuilding the final
URI, the current code inserts a '/' by itself between the hostname and
the path, which results in URIs with an extra '/':
https://ovirt.example.com//some/path/api

This is not only cosmetic as this can cause issues with cookie handling
if libgovirt accesses //some/path/api while the cookie is set for
/some/path/api.

Resolves: rhbz#1127156
(cherry picked from commit 914f75ee8d2264cb2fd4276a3a935f2f7bbfbc82)
---
 src/remote-viewer.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/remote-viewer.c b/src/remote-viewer.c
index 2d398c8..26f0634 100644
--- a/src/remote-viewer.c
+++ b/src/remote-viewer.c
@@ -656,6 +656,7 @@ parse_ovirt_uri(const gchar *uri_str, char **rest_uri, char **name, char **usern
         xmlFreeURI(uri);
         return FALSE;
     }
+    g_return_val_if_fail(*uri->path == '/', FALSE);
 
     /* extract VM name from path */
     path_elements = g_strsplit(uri->path, "/", -1);
@@ -675,7 +676,7 @@ parse_ovirt_uri(const gchar *uri_str, char **rest_uri, char **name, char **usern
     /* build final URI */
     rel_path = g_strjoinv("/", path_elements);
     /* FIXME: how to decide between http and https? */
-    *rest_uri = g_strdup_printf("https://%s/%s/api/", uri->server, rel_path);
+    *rest_uri = g_strdup_printf("https://%s%s/api/", uri->server, rel_path);
     *name = vm_name;
     g_free(rel_path);
     g_strfreev(path_elements);
